node/sc/bridge_test.go

```diff
@@ -665,3 +665,310 @@ func TestExtendedBridgeAndCallbackERC721(t *testing.T) {
+	assert.NoError(t, err)
+	backend.Commit()
+	assert.Nil(t, bind.CheckWaitMined(backend, tx))
+
+	return backend, operator, tester, b, erc20, erc721, erc20Addr, erc721Addr
+}
+
+// TestBridgeContract_InitStatus checks initial lock status.
+func TestBridgeContract_InitStatus(t *testing.T) {
+	_, _, _, b, _, _, erc20Addr, erc721Addr := generateBridgeTokenTestEnv(t)
+
+	// initial value check
+	isLocked, err := b.LockedTokens(nil, erc20Addr)
+	assert.NoError(t, err)
+	assert.Equal(t, false, isLocked)
+
+	isLocked, err = b.LockedTokens(nil, erc721Addr)
+	assert.NoError(t, err)
+	assert.Equal(t, false, isLocked)
+
+	isLocked, err = b.IsLockedKLAY(nil)
+	assert.NoError(t, err)
+	assert.Equal(t, false, isLocked)
+}
+
+// TestBridgeContract_InitRequest checks the following:
+// - the request value transfer can be allowed after registering it.
+func TestBridgeContract_InitRequest(t *testing.T) {
+	backend, operator, tester, b, erc20, erc721, _, _ := generateBridgeTokenTestEnv(t)
+
+	// check to allow value transfer
+	tx, err := erc20.RequestValueTransfer(tester, big.NewInt(1), operator.From, big.NewInt(0), nil)
+	assert.NoError(t, err)
+	backend.Commit()
+	assert.Nil(t, bind.CheckWaitMined(backend, tx))
+
+	tx, err = erc721.RequestValueTransfer(tester, big.NewInt(1), operator.From, nil)
+	assert.NoError(t, err)
+	backend.Commit()
+	assert.Nil(t, bind.CheckWaitMined(backend, tx))
+
+	tester.Value = big.NewInt(1)
+	tx, err = b.RequestKLAYTransfer(tester, tester.From, big.NewInt(1), nil)
+	assert.NoError(t, err)
+	backend.Commit()
+	assert.Nil(t, bind.CheckWaitMined(backend, tx))
+	tester.Value = nil
+}
+
+// TestBridgeContract_TokenLock checks the following:
+// - the token can be lock to prevent value transfer requests.
+func TestBridgeContract_TokenLock(t *testing.T) {
+	backend, operator, tester, b, erc20, erc721, erc20Addr, erc721Addr := generateBridgeTokenTestEnv(t)
+
+	// lock token
+	tx, err := b.LockToken(operator, erc20Addr)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusSuccessful, t)
+
+	tx, err = b.LockToken(operator, erc721Addr)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusSuccessful, t)
+
+	tx, err = b.LockKLAY(operator)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusSuccessful, t)
+
+	// check value after locking
+	isLocked, err := b.LockedTokens(nil, erc20Addr)
+	assert.NoError(t, err)
+	assert.Equal(t, true, isLocked)
+
+	isLocked, err = b.LockedTokens(nil, erc721Addr)
+	assert.NoError(t, err)
+	assert.Equal(t, true, isLocked)
+
+	isLocked, err = b.IsLockedKLAY(nil)
+	assert.NoError(t, err)
+	assert.Equal(t, true, isLocked)
+
+	// check to prevent value transfer
+	tx, err = erc20.RequestValueTransfer(tester, big.NewInt(1), operator.From, big.NewInt(0), nil)
+	assert.NoError(t, err)
+	backend.Commit()
+	assert.NotNil(t, bind.CheckWaitMined(backend, tx))
```

Comment:
How can we know whether the transfer is prevented?

Reply:
This `CheckWaitMined` will return error when the tx is reverted.